.github/workflows/build.yml

```diff
@@ -140,7 +143,7 @@ jobs:
         uses: codecov/codecov-action@v4
 
   go-semantic-release:
-    if: ${{ github.ref == 'refs/heads/main' }}
+    if: ${{ github.ref == 'refs/heads/main' }} || github.actor != 'liatrio-otel-collector-release[bot]'
```

Comment:
Is this condition correct? My thinking is no. Go semantic release should only run when `if the ref is main AND the GitHub action IS NOT the bot` right?

Reply:
I think this step name is misleading, because this is the multimod job. This job never runs a semantic release

Reply:
I renamed the job to "prepare-go-semantic-release" and added in the && statement